src/goto-symex/goto_symex_state.cpp

```diff
@@ -342,6 +342,9 @@ void goto_symex_statet::assignment(
   assert_l2_renaming(lhs);
   assert_l2_renaming(rhs);
 
+  if(is_shared && lhs.type().id() == ID_pointer)
```

Comment:
Add a comment here referencing an issue that describes what's wrong, and what we'd need to do to fix it.

Reply:
Done.